src/fixedpoint.ts

```diff
@@ -1,7 +1,21 @@
+/**
+ * Fixed point number implementation.
+ *
+ * This is useful for representing token amounts and other big numbers.
+ */
 export class FixedInt<T extends number> {
   _value: bigint;
+  /**
+   * The number of decimals to use.
+   */
   decimals: T;
```

Comment:
Should `decimals` have `protected` access instead of inexplicit `public`? Is it expected to allow users to update this after instantiation? Does decimals even need to be exposed

Reply:
i think its useful (necessary?) to expose. e.g. you might need only decimals for a calc e.g. https://github.com/tomquirk/juice-hooks/blob/6923900948383efcc300fdd5a529e65980d94101/src/utils/token.ts#L71-L72

But should be immutable.

Reply:
Ok cool! I think just make the setter protected so inheritance is still allowed to mess with it